### DISCUSSION TITLE
Deploy Next app to AWS Amplify

### DISCUSSION THREAD

**Discussion by hloughrey** [nrwl/nx#6772] [General]
I'm having difficulty deploying an Nx based Next app to AWS Amplify. We're currently pretty bought into the AWS infrastructure so can't use something like Vercel or Heroku. I could look to build & deploy the application within a Docker container, but this feels like overkill given that Amplify supports NextJs deployments (with the added benefit of a few optimisations).

The problem I'm getting is that AWS Amplify examines the package.json to identify if you're deploying an SSR or Static Next site & expects a `next build` script. While I could copy the root level package.json into the app directory, I wouldn't be able to build the app if it was making use of an Nx library.

Has anyone managed to deploy an Nx based Next app to Amplify? If so I'd love some pointers or if anyone else is struggling with the same issue I'd be happy to pair up. 

Cheers

**Comment by paulfalgout** [comment]
I haven't been able to try this yet, but it's relevant: https://docs.aws.amazon.com/amplify/latest/userguide/server-side-rendering-amplify.html

**Reply by hloughrey** [reply]
This doesn't work unfortunately as the automatic deployments via Amplify query the package.json to know how to deploy the app.
